src/include/duckdb/parser/parsed_data/create_index_info.hpp

```diff
@@ -24,24 +24,32 @@ struct CreateIndexInfo : public CreateInfo {
 	IndexType index_type;
 	//! Name of the Index
 	string index_name;
-	//! If it is an unique index
-	bool unique = false;
+	//! Index Constraint Type
+	IndexConstraintType constraint_type;
 	//! The table to create the index on
 	unique_ptr<BaseTableRef> table;
 	//! Set of expressions to index by
 	vector<unique_ptr<ParsedExpression>> expressions;
+	//! These should never be bound
+	vector<unique_ptr<ParsedExpression>> parsed_expressions;
```

Comment:
What is the difference between this and `expressions`?

Reply:
The difference is that expressions eventually go through the binder, which doesn't allow us to serialize them.
Maybe renaming these to unbound_expressions?

Reply:
Can't you just copy them in the binder so that they aren't lost there, rather than storing a second copy in the `create_index_info`?